src/main/java/seedu/address/logic/parser/TagCommandGroupParser.java

```diff
@@ -40,12 +44,28 @@ public TagCommandGroup parse(String args) throws ParseException {
         switch (commandSpecifier) {
         case CreateTagCommand.COMMAND_SPECIFIER:
             String[] argsToPass = Arrays.copyOfRange(argArray, 1, argArray.length);
-            return new CreateTagCommandParser().parse(String.join(" ", argsToPass));
```

Comment:
I think we should keep the `CreateTagCommandParser` and `DeleteTagCommandParser` here, to maintain a single layer of abstraction within this `switch` block?

Reply:
I change it out as I didn't want to duplicate code. I didn't think of SLAP when I changed it. Perhaps I could abstract the creation into another function?
```java
switch (commandSpecifier) {
        case CreateTagCommand.COMMAND_SPECIFIER:
            return parseCreateTagCommand(argArray);
        case DeleteTagCommand.COMMAND_SPECIFIER:
            return parseDeleteTagCommand(argArray);
        case RemoveTagCommand.COMMAND_SPECIFIER:
            return parseRemoveTagCommand(argArray);
        default:
            return new TagCommandParser().parse(trimmedArgs);
        }
    }

private CreateTagCommand parseCreateTagCommand(String[] argArray) throws ParseException {
    String[] argsToPass = Arrays.copyOfRange(argArray, 1, argArray.length);
    return new CreateTagCommand(parseArgs(argsToPass));
}

private DeleteTagCommand parseDeleteTagCommand(String[] argArray) throws ParseException {
    String[] argsToPass = Arrays.copyOfRange(argArray, 1, argArray.length);
    return new DeleteTagCommand(parseArgs(argsToPass));
}

private RemoveTagCommand parseRemoveTagCommand(String[] argArray) throws ParseException {
    String[] argsToPass = Arrays.copyOfRange(argArray, 1, argArray.length);
    return new RemoveTagCommandParser().parse(String.join(" ", argsToPass));
}
```

Reply:
Perhaps we can do
```
String[] argsToPass = Arrays.copyOfRange(argArray, 1, argArray.length);
switch (commandSpecifier) {
case CreateTagCommand.COMMAND_SPECIFIER:
    return new CreateTagCommand(parseArgs(argsToPass));
case DeleteTagCommand.COMMAND_SPECIFIER:
    return new DeleteTagCommand(parseArgs(argsToPass));
case RemoveTagCommand.COMMAND_SPECIFIER:
    return new RemoveTagCommandParser().parse(String.join(" ", argsToPass));
default:
    return new TagCommandParser().parse(trimmedArgs);
}
```

Reply:
"we"??

> 



> ```java
> String[] argsToPass = Arrays.copyOfRange(argArray, 1, argArray.length);
> ```
I thought of this as well. Wouldn't there be an error if the array is only size 1?

Reply:
As below,
![image](https://user-images.githubusercontent.com/14901260/196035664-16339a25-743c-402a-a6f5-ab06e02d888e.png)